src/DSE.Open.Values/AgeInMonths.cs

```diff
@@ -1,201 +1,184 @@
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(months, MaxMonths);
+
+        var maxMonths = int.MaxValue - (years * 12);
+
+        if (months > maxMonths)
+        {
+            throw new ArgumentOutOfRangeException(nameof(years),
+                "The resulting value is outside the range of a AgeInMonths");
+        }
+
+        checked
+        {
+            TotalMonths = (years * 12) + months;
+        }
     }
 
     public AgeInMonths(int totalMonths)
     {
+        ArgumentOutOfRangeException.ThrowIfNegative(totalMonths);
         TotalMonths = totalMonths;
     }
 
     /// <summary>
-    /// The number of complete years represented by the value.
+    /// The total number of months represented by the value.
     /// </summary>
-    public int Years => TotalMonths / 12;
+    public int TotalMonths { get; }
 
     /// <summary>
-    /// The number of months represented by the value in addition to the <see cref="Years"/>.
+    /// The total number of years and months represented by the value.
     /// </summary>
-    public int Months => TotalMonths % 12;
+    /// <example>
+    /// 13 months would return 1 year and 1 month.
+    /// </example>
+    public (int Years, int Months) YearsAndMonths()
+    {
+        var years = TotalMonths / 12;
+        var months = TotalMonths - (years * 12);
+        return (years, months);
+    }
 
     /// <summary>
-    /// The total number of months represented by the value. This is the combined value of
-    /// <see cref="Years"/> and <see cref="Months"/>.
+    /// Adds the specified number of months to the current value.
     /// </summary>
-    public int TotalMonths { get; }
-
+    /// <param name="months"></param>
+    /// <returns></returns>
+    /// <exception cref="OverflowException">The resulting value is outside the range of a <see cref="AgeInMonths"/>.</exception>
     public AgeInMonths AddMonths(int months)
     {
-        return new(TotalMonths + months);
+        checked
+        {
+            return new AgeInMonths(TotalMonths + months);
+        }
     }
 
+    /// <summary>
+    /// Adds the specified number of years to the current value.
+    /// </summary>
+    /// <param name="years"></param>
+    /// <returns></returns>
+    /// <exception cref="OverflowException">The resulting value is outside the range of a <see cref="AgeInMonths"/>.</exception>
     public AgeInMonths AddYears(int years)
     {
-        return new(TotalMonths + (years * 12));
+        checked
+        {
+            return new AgeInMonths(TotalMonths + (years * 12));
+        }
     }
 
     public int CompareTo(AgeInMonths other)
     {
         return TotalMonths.CompareTo(other.TotalMonths);
     }
 
-    public bool TryFormat(Span<char> destination, out int charsWritten)
+    public static bool IsValidValue(int value)
     {
-        return TryFormat(destination, out charsWritten, default, default);
+        return value >= 0;
```

Comment:
Should we have a more sensible upper bound?

Reply:
Perhaps `&& < 1200` ?